### PR TITLE
root zone fixes in core and gsql backends

### PR DESCRIPTION
**Pull request by mind04**
Core and gsql backends are able to serve/AXFR a root/tld zone
Bind backend still need some love (in a separate pull)

### CODE REVIEW

**Comment by Habbie** [comment]
This branch has conflicts

File: regression-tests.rootzone/tests/direct-ns/expected_result [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+1	net.	IN	NS	172800	d.gtld-servers.net.
+1	net.	IN	NS	172800	e.gtld-servers.net.
+1	net.	IN	NS	172800	f.gtld-servers.net.
+1	net.	IN	NS	172800	g.gtld-servers.net.
+1	net.	IN	NS	172800	h.gtld-servers.net.
+1	net.	IN	NS	172800	i.gtld-servers.net.
+1	net.	IN	NS	172800	j.gtld-servers.net.
+1	net.	IN	NS	172800	k.gtld-servers.net.
+1	net.	IN	NS	172800	l.gtld-servers.net.
+1	net.	IN	NS	172800	m.gtld-servers.net.
```
Comment (Habbie) [inline regression-tests.rootzone/tests/direct-ns/expected_result]: If your ROOT file only has two NSes in each set, why are we getting 13 here?

Comment (Habbie) [inline regression-tests.rootzone/tests/direct-ns/expected_result]: Nevermind, later commit fixes this!

**Comment by Habbie** [comment]
This is awesome work from both of you! Happy to merge this when the conflicts are gone.

**Comment by cmouse** [comment]
" 	AXFR-out rectify: don't set odername for empty non-terminals only der… … 	", typo fix if you are rebasing

**Pull request merged**